### DISCUSSION TITLE
Removing installation manually when uninstaller is not working

### DISCUSSION THREAD

**Discussion by cz-dev-ge** [microsoft/winget-cli#1386] [Q&A] [answered]
# Brief description of your issue

Is it possible to remove an application manually from winget? I tried to uninstall an app, where I cannot run the uninstaller, therefore uninstall fails, but I want to get rid of it.

# Steps to reproduce

Rename an uninstaller exe and try to uninstall with winget.

# Expected behavior

Any means of removing the application from winget so it does not know about that application anymore.

# Actual behavior

Uninstall fails. App entry stays.

# Environment

Nothing special.

**Comment by jedieaston** [comment, marked as answer]
WinGet doesn't keep track of where apps are putting files by itself, so it's not possible at the moment for it to nuke an app outside of using the registered uninstaller. The Add and Remove Programs table offers to remove an app from the table if the uninstaller is missing, maybe that's something WinGet should be able to do.

**Reply by keenbowl3009** [reply]
> WinGet doesn't keep track of where apps are putting files by itself, so it's not possible at the moment for it to nuke an app outside of using the registered uninstaller. The Add and Remove Programs table offers to remove an app from the table if the uninstaller is missing, maybe that's something WinGet should be able to do.

So what winget gets done the worst of both worlds, with regards to package management with the shell?

**Comment by TrashAccount3424** [comment]
Not related to winget but usually if a program can not be uninstalled you can just manually remove (or rename) the registry folder that corresponds to the program from 

`\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\Installer\Products\`

Then install the program and uninstall the program once to clean up.

**Reply by EqualDust** [reply]
If have to manually remove things, then why tf does winget exist?

**Comment by EqualDust** [comment]
Why was the issue closed? Trusting winget means that my computer will get filled with garbage. That issue was precisely why people were asking for an uninstaller.
